<launch>

  <arg name="image_raw_topic" default="/usb_cam/image_raw"/>
  <arg name="twist_cmd_topic" default="/cmd_vel"/>
  <arg name="base_folder" default="/set1"/>

  <node name="rosserial" pkg="rosserial_python" type="serial_node.py" args="tcp" output="screen" />

  <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
  </node>

    <node name="write_data" pkg="ackerbot" type="mlwrite_data_lanefollow.py" output="screen">
      <param name="image_raw_topic" value="$(arg image_raw_topic)"/>
      <param name="twist_cmd_topic" value="$(arg twist_cmd_topic)"/>
      <param name="base_folder" value="$(arg base_folder)"/>
    </node>

</launch>
